packages/flutter/lib/src/material/paginated_data_table.dart

```diff
@@ -426,74 +426,79 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
+                    child: Padding(
+                      padding: EdgeInsetsDirectional.only(start: startPadding, end: 14.0),
+                      child: Row(
+                        mainAxisAlignment: MainAxisAlignment.end,
+                        children: headerWidgets,
+                      ),
                     ),
                   ),
                 ),
               ),
             ),
-          ),
-          SingleChildScrollView(
-            scrollDirection: Axis.horizontal,
-            dragStartBehavior: widget.dragStartBehavior,
-            child: DataTable(
-              key: _tableKey,
-              columns: widget.columns,
-              sortColumnIndex: widget.sortColumnIndex,
-              sortAscending: widget.sortAscending,
-              onSelectAll: widget.onSelectAll,
-              dataRowHeight: widget.dataRowHeight,
-              headingRowHeight: widget.headingRowHeight,
-              horizontalMargin: widget.horizontalMargin,
-              columnSpacing: widget.columnSpacing,
-              showCheckboxColumn: widget.showCheckboxColumn,
-              rows: _getRows(_firstRowIndex, widget.rowsPerPage),
-            ),
-          ),
-          DefaultTextStyle(
-            style: footerTextStyle,
-            child: IconTheme.merge(
-              data: const IconThemeData(
-                opacity: 0.54
+            SingleChildScrollView(
+              scrollDirection: Axis.horizontal,
+              dragStartBehavior: widget.dragStartBehavior,
+              child: ConstrainedBox(
+                  constraints: BoxConstraints(minWidth: constraints.minWidth),
+                  child: DataTable(
+                    key: _tableKey,
+                    columns: widget.columns,
+                    sortColumnIndex: widget.sortColumnIndex,
+                    sortAscending: widget.sortAscending,
+                    onSelectAll: widget.onSelectAll,
+                    dataRowHeight: widget.dataRowHeight,
+                    headingRowHeight: widget.headingRowHeight,
+                    horizontalMargin: widget.horizontalMargin,
+                    columnSpacing: widget.columnSpacing,
+                    showCheckboxColumn: widget.showCheckboxColumn,
+                    rows: _getRows(_firstRowIndex, widget.rowsPerPage),
+                ),
               ),
-              child: Container(
-                height: 56.0,
-                child: SingleChildScrollView(
-                  dragStartBehavior: widget.dragStartBehavior,
-                  scrollDirection: Axis.horizontal,
-                  reverse: true,
-                  child: Row(
-                    children: footerWidgets,
+            ),
+            DefaultTextStyle(
+              style: footerTextStyle,
+              child: IconTheme.merge(
+                data: const IconThemeData(
+                  opacity: 0.54
+                ),
+                child: Container(
+                  height: 56.0,
```

Comment:
this looks like it won't handle text zoom correctly, fwiw

Reply:
Filed https://github.com/flutter/flutter/issues/48522